### DISCUSSION TITLE
What's the hosting architecture of the web client ?

### DISCUSSION THREAD

**Discussion by adam-devel** [dani-garcia/vaultwarden#5295] [Q&A] [answered]
Should I expect the web client to be hosted by vaultwarden ? or am I expected to host it with a simple file server ? or does it come with it's own dedicated server software ?

A file server seems too simple, and the use of nodejs suggests that the client uses a dedicated nodejs server, but when I try to go to the import page in the android client, it redirects me to `{valwarden's address}/#/tools/import`

the client expects the imprt tool (which is part of the webclient) to be hosted at the same location of the backend by the server.

so either, vaultwarden is expected to take care of hosting the web client (in which case, I would expect a location in which I am expected to put the server files), or I am expected to do some networking magic in order to forward some requests to `vaultwarden` and others (such as `/toos/*` to the sever hosting the webclient

**TLDR**: How does the hosting architecture look like ? I get different signals from different resources and I am struggling to reconcile them.

**Comment by BlackDex** [comment]
Checked the wiki already? https://github.com/dani-garcia/vaultwarden/wiki#supported-features

TLDR: Vaultwarden is an all-in-one system when using the container images. if not using container images, then some more steps are needed, for that, you need to check the documentation of those deployments.

**Reply by adam-devel** [reply, marked as answer]
> Vaultwarden is an all-in-one system when using the container images. if not using container images, then some more steps are needed, for that, you need to check the documentation of those deployments

You are mostly describing the packaging strategies there.
Saying "Vaultwarden is an all-in-one system" isn't very helpful. It doesn't tell me what "all" includes.

To answer my own question from what i understand so far, it seems that vaultwarden (and bitwarden) play a double role: 1) Vaultwarden is hosting the backend, accessible through an API. and 2) Vaultwarden is also hosting a web client consuming said API.

Unlike the mobile client, the web client isn't a standalone component.

Edit: it is possible to host the web client separately if you need to do so and are willing to bother with dispatching requests correctly. but that's not the expectation for most cases.

**Reply by BlackDex** [reply]
That depends on how you few it. The web-vault is a a standalone component, and if really really wanted, you can even use nginx or apache to host those files. Vaultwarden doesn't do anything more then what nginx or apache would do by server those static files.

**Reply by adam-devel** [reply]
Understood, strictly speaking you can pull any part of an open source program out and make it standalone tho. That's why i mostly talked about expectations. the fact that the mobile app redirects to `{valwarden's address}/#/tools/import` and the fact that vaultwarden bothers with hosting a simple static app shows the expectation is to serve the web client by vaultwarden
